### DISCUSSION TITLE
Prevent video player from entering "pop out" mode when switching apps

### DISCUSSION THREAD

**Discussion by seanmaisch** [TheWidlarzGroup/react-native-video#4979] [Ideas]
### What problem are you trying to solve?

When a user navigates away from our app, whether it be by switching apps manually or externally linking to the web, the video player enters "pop out" mode, which can be construed by some as `pictureInPicture`.

We would really like to prevent iOS from popping the player out when leaving the application.

### Proposed solution

A way to prevent the video player from entering "pop out" mode when the app loses focus / the user switches apps, without having to set `pictureInPicture={false}` (which does not rectify this issue) or completely unload the player.

### Alternatives considered

- Simply setting `pictureInPicture={false}` does not rectify this issue.
- Completely unloading the player when the app loses focus or is inactive. This was considered but the author would like to avoid it due to performance issues and poor user experience ("that seems like a nuclear option").

### Platforms

iOS

### Additional context

Referenced as well by @jcksncllwy in https://github.com/TheWidlarzGroup/react-native-video/issues/2373#issuecomment-1537235603

**Comment by KrzysztofMoch** [comment]
So player is going into Picture in Picture even if `pictureInPicture` is set to false? - this is unexpected

**Comment by seanmaisch** [comment]
Correct. In the example mentioned by @jcksncllwy in https://github.com/TheWidlarzGroup/react-native-video/issues/2373#issuecomment-1537235603, they are attempting to use the video player as a background element and when switching the application, it opens the video player in PIP.

My use-case where the issue is presenting itself is as follows:

We have a "preview" video for users who are not paid users. They are only allowed to watch for a certain amount of time, but if they switch away from the application (via ExternalLink) to convert to a paid user, the video follows them outside of the application and they are able to continue to watch in the "pop out" / "picture-in-picture" mode indefinitely, which is problematic for us.

[a comment by an automated account is not shown]

**Comment by TWG-AI** [comment]
In v7 the `autoEnterPictureInPicture` prop on `<VideoView>` (defaults to `false`) controls whether iOS automatically enters PiP when the user switches apps. The default already prevents pop-out:

```tsx
<VideoView
  player={player}
  pictureInPicture={false}          // no PiP button
  autoEnterPictureInPicture={false} // don't pop out automatically
/>
```

> I'm an AI — iOS PiP behavior can vary; verify against the v7 docs.
